gpkit/constraints/link.py

```diff
@@ -0,0 +1,57 @@
+"Implements LinkConstraint"
+from .set import ConstraintSet
+from ..varkey import VarKey
+from .. import SignomialsEnabled
+
+
+class LinkConstraint(ConstraintSet):
+    """A ConstraintSet that links duplicate variables in its constraints
+
+    Variables with the same name are linked
+
+    Arguments
+    ---------
+    constraints: iterable
+        valid argument to ConstraintSet
+    include_only: set
+        whitelist of variable names to include
+    exclude: set
+        blacklist of variable names, supercedes include_only
+    """
+    def __init__(self, constraints, include_only=None, exclude=None):
+        ConstraintSet.__init__(self, constraints)
+        varkeys = self.varkeys
+        linkable = set()
+        for varkey in varkeys:
+            if len(varkeys[varkey.name]) > 1:
+                linkable.add(varkey.name)
+        if include_only:
+            linkable &= set(include_only)
+        if exclude:
+            linkable -= set(exclude)
+        self.linked, self.reverselinks = {}, {}
+        for name in linkable:
+            vks = varkeys[name]
+            descr = dict(vks[0].descr)
+            value = descr.pop("value", None)
```

Comment:
Just delete the key, don't get the value. If there was a value it'll be in `self.substitutions` below.